Duplicate Orphan Instances cause build failure

When trying to build, I get the following error:
```
src/Language/REST/Types.hs:72:10: error:
    Duplicate instance declarations:
      instance Hashable a => Hashable (OS.Set a)
        -- Defined at src/Language/REST/Types.hs:72:10
      instance Hashable v => Hashable (OS.Set v)
        -- Defined in ‘hashable-1.3.5.0:Data.Hashable.Class’
   |
72 | instance Hashable a => Hashable (OS.Set a) where
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Language/REST/Types.hs:75:10: error:
    Duplicate instance declarations:
      instance (Hashable a, Hashable b) => Hashable (M.Map a b)
        -- Defined at src/Language/REST/Types.hs:75:10
      instance (Hashable k, Hashable v) => Hashable (M.Map k v)
        -- Defined in ‘hashable-1.3.5.0:Data.Hashable.Class’
   |
75 | instance (Hashable a, Hashable b) => Hashable (M.Map a b) where
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

It is pretty much the same issue as seen in #3 and  #1.

You should not define these orphan instances without using a newtype wrapper.  This way ghc will be able to distinguish between the different implementations and choose the correct one for your library.

--- a/rest-rewrite.cabal
+++ b/rest-rewrite.cabal
@@ -51,7 +51,7 @@ library
   hs-source-dirs: src
   build-depends:  base                 >= 4.7 && < 5
                 , containers           >= 0.6.2 && < 0.7
-                , hashable             >= 1.3.0 && < 1.4
+                , hashable             >= 1.3.4 && < 1.4
                 , process              >= 1.6.9 && < 1.7
                 , parsec               >= 3.1.14 && < 3.2
                 , mtl                  >= 2.2.2 && < 2.3

--- a/src/Language/REST/Types.hs
+++ b/src/Language/REST/Types.hs
@@ -6,7 +6,7 @@
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE RankNTypes #-}
-{-# OPTIONS_GHC -Wno-orphans #-}
+
 
 module Language.REST.Types (
     prettyPrint
@@ -69,11 +69,6 @@ instance Show Relation where
   show GTE = "≥"
   show EQ  = "≅"
 
-instance Hashable a => Hashable (OS.Set a) where
-  hashWithSalt i s = hashWithSalt i (OS.toList s)
-
-instance (Hashable a, Hashable b) => Hashable (M.Map a b) where
-  hashWithSalt i s = hashWithSalt i (M.toList s)
 
 toOrderedSet :: (Eq a, Hashable a, Ord a) => S.HashSet a -> OS.Set a
 toOrderedSet = OS.fromList . S.toList

--- a/src/Language/REST/Types.hs
+++ b/src/Language/REST/Types.hs
@@ -22,7 +22,6 @@ import           Data.Hashable
 import qualified Data.List as L
 import qualified Data.HashSet as S
 import qualified Data.Set as OS
-import qualified Data.Map as M
 import qualified Data.Text as T
 import           Text.Printf
 

--- a/stack.yaml
+++ b/stack.yaml
@@ -2,6 +2,9 @@ resolver: lts-18.24
 packages:
   - .
 
+extra-deps:
+  - hashable-1.3.5.0
+
 nix:
   packages: [graphviz, z3]
   path: [nixpkgs=./nixpkgs.nix]

--- a/stack.yaml.lock
+++ b/stack.yaml.lock
@@ -3,7 +3,14 @@
 # For more information, please see the documentation at:
 #   https://docs.haskellstack.org/en/stable/lock_files
 
-packages: []
+packages:
+- completed:
+    hackage: hashable-1.3.5.0@sha256:3a2beeafb220f9de706568a7e4a5b3c762cc4c9f25c94d7ef795b8c2d6a691d7,4240
+    pantry-tree:
+      size: 1248
+      sha256: 4df2f6b536a0fcc5f7d562cb29e373f27dc4a2747452ac5cc74c1599cab22fc5
+  original:
+    hackage: hashable-1.3.5.0
 snapshots:
 - completed:
     size: 587821

